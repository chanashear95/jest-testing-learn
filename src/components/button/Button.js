function Button({label}){
    return(
        <button className="btn-style" data-testid="button">
           Hi {label}
        </button>
    )
}

export default Button;