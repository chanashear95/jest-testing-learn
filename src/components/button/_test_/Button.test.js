import ReactDom from 'react-dom';
import Button from 'components/button/Button';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; 
import renderer from 'react-test-renderer';

afterEach(cleanup);
it('Renders without crashing', () => {
    let div = document.createElement('div');
    ReactDom.render(<Button />, div);
})

it('renders button correctly',  () => {
    const {getByTestId} = render(<Button label={"Click me please"}/>);
    expect(getByTestId('button')).toHaveTextContent('Click me please');
})

it('renders button correctly',  () => {
    const {getByTestId} = render(<Button label={"Save"}/>);
    expect(getByTestId('button')).toHaveTextContent('Save');
})

it('matches snapshot 1' , () => {
    const tree = renderer.create(<Button label={'Save'}/>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot 2' , () => {
    const tree = renderer.create(<Button label={'click me please'}/>).toJSON();
    expect(tree).toMatchSnapshot();
})