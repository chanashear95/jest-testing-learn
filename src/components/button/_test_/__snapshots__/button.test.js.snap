// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<button
  className="btn-style"
  data-testid="button"
>
  Hi 
  Save
</button>
`;

exports[`matches snapshot 2 1`] = `
<button
  className="btn-style"
  data-testid="button"
>
  Hi 
  click me please
</button>
`;
